src/plugins/event-plugins/NavigationPlugin.ts

```diff
@@ -184,11 +199,24 @@ export class NavigationPlugin extends InternalPlugin {
     /**
      * W3C specification: https://www.w3.org/TR/navigation-timing-2/#bib-navigation-timing
      */
-    performanceNavigationEventHandlerTimingLevel2 = (entryData: any): void => {
+    performanceNavigationEventHandlerTimingLevel2 = (
+        entryData: PerformanceNavigationTiming
+    ): void => {
+        if (this.config.ignore(entryData)) {
+            return;
+        }
+
         const eventDataNavigationTimingLevel2: NavigationEvent = {
             version: '1.0.0',
-            initiatorType: entryData.initiatorType,
-            navigationType: entryData.type,
+            initiatorType: entryData.initiatorType as
+                | 'navigation'
+                | 'route_change',
+            navigationType: entryData.type as
```

Comment:
question: `PerformanceNavigationTiming.type` allows for types `navigation`, `reload`, `back_forward`, and `pre-render`. Where does `reserved` come from? 

https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming/type